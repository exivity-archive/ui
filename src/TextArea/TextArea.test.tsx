import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import { TextArea } from '.'

test('TextArea snapshot', () => {
  const textArea = renderer.create(<TextArea onChange={jest.fn()} />)
  expect(textArea).toMatchSnapshot()
})

test('value gets changed after an on change event', () => {
  const initialValue = 'initialValue'
  const updatedValue = 'changed'

  const onChangeMock = jest.fn(x => x)

  const textArea = mount(<TextArea value={initialValue} onChange={onChangeMock} />)

  textArea
    .find('textarea')
    .simulate('change', { target: { value: updatedValue } })

  expect(onChangeMock.mock.results[0].value).toBe(updatedValue)
})
