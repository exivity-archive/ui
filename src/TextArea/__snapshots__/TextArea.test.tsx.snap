// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea snapshot 1`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  font-size: 16px;
  display: block;
  box-sizing: border-box;
  width: 100%;
  padding: calc(0.5em - 1px) 0.5em;
  border-radius: 4px;
  outline: 0;
  border: 0;
  border: 1px solid #eee;
  background-color: #eee;
  --focus-color: 153,153,153;
}

.c0:hover {
  border-bottom: 1px solid rgba(var(--focus-color),0.5);
}

.c0:focus {
  border-bottom: 1px solid rgba(var(--focus-color),1);
}

.c0::-webkit-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0::-moz-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0:-ms-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0::placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0[disabled] {
  cursor: not-allowed;
  box-shadow: inset 0 0 999em rgba(128,128,128,0.2);
}

<textarea
  className="c0"
  onChange={[Function]}
  secondary={true}
  theme={
    Object {
      "colours": Object {
        "black": "#000",
        "blue": "#00a8d8",
        "dark": "#222",
        "gray": "#999",
        "green": "#4caf50",
        "lightGray": "#eee",
        "red": "#ef5350",
        "white": "#fff",
        "yellow": "#fff68c",
      },
      "fonts": Object {
        "custom": Object {
          "families": Array [],
          "urls": Array [],
        },
        "google": Object {
          "families": Array [
            "Fira Mono:400",
            "Fira Sans:400,600",
          ],
        },
      },
      "global": Object {
        "baseSize": 16,
        "borderRadius": 4,
        "borderWidth": 1,
        "fontFamily": "'Fira Sans', sans-serif",
        "fontFamilyMonospace": "'Fira Mono', monospace",
        "lineHeight": 1.5,
        "outlineAlpha": 0.15,
        "outlineWidth": 4,
        "purposes": Object {
          "danger": "#ef5350",
          "primary": "#00a8d8",
          "secondary": "#999",
          "success": "#4caf50",
        },
        "shadowAlpha": 0.1,
        "sizes": Object {
          "large": 1.1875,
          "small": 0.875,
        },
        "spacing": 1.25,
        "textColor": "#222",
        "textColorMuted": "#999",
      },
    }
  }
  value=""
/>
`;
